File: ABC220/d.py
# -*- coding: utf-8 -*-
def tmp_fun(x, y, num):
    hoge = (x+y)%10
    fuga = (x*y)%10
    if(num == len(A)):
        ans[hoge] +=1
        ans[fuga] +=1
    else:

        return tmp_fun(hoge, A[num], num+1), tmp_fun(fuga, A[num], num+1)


n = int(input())
A = list(map(int, input().split()))

ans = [0]*10

tmp_fun(A[0], A[1], 2)

for i in range(10):
    print(ans[i])

File: ABC220/d.cc
#include <iostream>
#include <cmath>
#include <algorithm>
#include <vector>
#include <cassert>
#include <numeric>
#include <string>
#include <map>
using namespace std;

int tmp_fun(long int A[], long int a, long int b, long int num, long int size);

int ans[10] ={0};

int tmp_fun(long int A[], long int a, long int b, long int num, long int size){
  int hoge, fuga;
  hoge = (a+b)%10;
  fuga = (a*b)%10;
  if(num == size){
    ans[hoge] += 1;
    ans[fuga] += 1;
    return 0;
  }else{
    return tmp_fun(A, A[hoge], A[fuga], 2+1, size);
  }

}

int main(){
  long int n, i;
  std::cin >> n;
  long int A[n];
  for(i=0; i<n; i++){
    std::cin >> A[i];
  }

  tmp_fun(A, A[0], A[1], 2, n);
  for(i=0; i<10; i++){
    std::cout << ans[i] << '\n';
  }

  return 0;
}
